packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx

```diff
@@ -1,413 +1,162 @@
-        { searchBox }
-        <FocusZone ref='focusZone' direction={ FocusZoneDirection.horizontal } role='menubar' >
-          <div className={ css('ms-CommandBar-primaryCommands', styles.primaryCommands) } ref='commandSurface'>
-            { renderedItems.map((item, index) => (
-              this._renderItemInCommandBar(item, index, expandedMenuItemKey)
-            )).concat((renderedOverflowItems && renderedOverflowItems.length) ? [
-              <div className={ css('ms-CommandBarItem', styles.item) } key={ OVERFLOW_KEY } ref={ OVERFLOW_KEY }>
-                <button
-                  type='button'
-                  id={ this._id + OVERFLOW_KEY }
-                  className={ css('ms-CommandBarItem-link', styles.itemLink, {
-                    ['is-expanded ' + styles.itemLinkIsExpanded]: (expandedMenuItemKey === OVERFLOW_KEY)
-                  }) }
-                  onClick={ this._onOverflowClick }
-                  role='menuitem'
-                  aria-label={ this.props.elipisisAriaLabel || '' }
-                  aria-haspopup={ true }
-                  data-automation-id='commandBarOverflow'
-                >
-                  <Icon className={ css('ms-CommandBarItem-overflow', styles.itemOverflow) } iconName='more' />
-                </button>
-              </div>
-            ] : []) }
-          </div>
-          <div className={ css('ms-CommandBar-sideCommands', styles.sideCommands) } ref='farCommandSurface'>
-            { renderedFarItems.map((item, index) => (
-              this._renderItemInCommandBar(item, index, expandedMenuItemKey, true)
-            )) }
-          </div>
-        </FocusZone>
-        { (contextualMenuProps) ?
-          (<ContextualMenu
-            className={ css('ms-CommandBar-menuHost') }
-            isBeakVisible={ true }
-            directionalHint={ DirectionalHint.bottomAutoEdge }
-            { ...contextualMenuProps }
-            targetElement={ contextualMenuTarget }
-            labelElementId={ expandedMenuId }
-            onDismiss={ this._onContextMenuDismiss }
-          />
-          ) : (null) }
-      </div>
+      <ResizeGroup
+        data={ commandBardata }
+        onReduceData={ this._onReduceData }
+        onRenderData={ (data: ICommandBarData) => {
+          return (
+            <div className={ css('ms-CommandBar', styles.root) }>
+
+              {/*Optional Search*/ }
+              { isSearchBoxVisible &&
+                this._onRenderSearch(this.props)
+              }
+
+              {/*Primary Items*/ }
+              <OverflowSet
+                className={ css(styles.primarySet) }
+                items={ data.primaryItems }
+                overflowItems={ data.overflowItems.length ? data.overflowItems : null }
+                onRenderItem={ onRenderItems }
+                onRenderOverflowButton={ (renderedOverflowItems: ICommandBarItemProps[]) => {
```

Comment:
shouldn't the default button be good enough here? Do you really need to implement an override?